### ISSUE TITLE
Question about segmentation

### ISSUE THREAD

**Issue by ChanningChe** [arahusky/Tensorflow-Segmentation#17]
Hi,
I'm trying to build a simple street view image segmentation program with tensorflow. 
I take your program for reference and build a toy neural network as below.
I don't use max pool in my program yet, the input img are all (720,960,3) and there are 32 final classes.
The model below contains: 
* 4 convolution layer with stride=2
* 1 convolution layer with stride=1
* 4 deconvolution layer with stride=2
But the result I get is weird that all the pixels are classified into the same class.
And every pixel is classified as the class whose number is the most in the img.
For example, in a given img, if the building pixels are the most, then all pixels will be classified into Building class. So my final result is always the same colour. I doubt my model is still doing some image classification work, not segmentation. I mean it is trying to classify all pixels into the same class which can get the minimum loss, not classify every pixel independently.

I don't know whether I have any misunderstanding of segmentation?
Could you pls give some advice?

Thanks a lot~

# encoder part:
		w1 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,self.IMG_CHANNEL,64])
		b1 = self.get_bias([64])
		conv1 = self.conv2d_layer(self.X_ph,w1,b1,stride=2)
		print (conv1.shape) #(?, 360, 480, 64)

		w2 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,64,64])
		b2 = self.get_bias([64])
		conv2 = self.conv2d_layer(conv1,w2,b2,stride=2)
		# pool1 = self.max_pool_layer(conv2)
		print (conv2.shape) #(?, 180, 240, 64)

		w3 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,64,64])
		b3 = self.get_bias([64])
		conv3 = self.conv2d_layer(conv2,w3,b3,stride=2)
		print (conv3.shape) #(?, 90, 120, 64)

		w4 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,64,64])
		b4 = self.get_bias([64])
		conv4 = self.conv2d_layer(conv3,w4,b4,stride=2)
		# pool2 = self.max_pool_layer(conv4)
		print (conv4.shape) #(?, 45, 60, 64)

# decoder part:
		w5 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,64,128])
		b5 = self.get_bias([128])
		deconv5 = self.conv2d_layer(conv4,w5,b5)
		print (deconv5.shape) #(?, 45, 60, 128)
		
		w6 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,128,128])
		b6 = self.get_bias([128])
		shape = tf.shape(conv3)
		deconv6 = self.deconv2d_layer(deconv5,w6,b6,tf.stack([shape[0],90,120,128]))
		print (deconv6.shape) #(?, 90, 120, 128)

		w7 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,128,128])
		b7 = self.get_bias([128])
		shape = tf.shape(conv2)
		deconv7 = self.deconv2d_layer(deconv6,w7,b7,tf.stack([shape[0],180,240,128]))
		print (deconv7.shape) #(?, 180, 240, 128)
		
		w8 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,128,128])
		b8 = self.get_bias([128])
		shape = tf.shape(conv1)
		deconv8 = self.deconv2d_layer(deconv7,w8,b8,tf.stack([shape[0],360,480,128]))
		print (deconv8.shape) #(?, 360, 480, 128)

		w9 = self.get_weight([self.FILTER_SIZE,self.FILTER_SIZE,self.CLASS_NUM,128])
		b9 = self.get_bias([self.CLASS_NUM])
		shape = tf.shape(self.X_ph)
		deconv9 = 
                self.deconv2d_layer(deconv8,w9,b9,tf.stack([shape[0],shape[1],shape[2],self.CLASS_NUM]))
		print (deconv9.shape) #(?, ?, ?, 32)

		self.predict = tf.nn.softmax(deconv9)
		self.loss = 
                 tf.reduce_mean(tf.losses.sparse_softmax_cross_entropy(labels=self.y_ph,logits=deconv9))
		self.optimizer = tf.train.AdamOptimizer(self.LEARNING_RATE,epsilon=0.1).minimize(self.loss)
